Source/OxyPlot/Series/BarSeries/RectangleBarSeries.cs

```diff
@@ -302,5 +324,75 @@ protected virtual bool IsValid(double v)
         {
             return !double.IsNaN(v) && !double.IsInfinity(v);
         }
-    }
+
+		/// <summary>
+		/// Find index of max(x) &lt;= target x in a list of data points
+		/// </summary>
+		/// <param name='items'>
+		/// vector of data points
+		/// </param>
+		/// <param name='targetX'>
+		/// target x.
+		/// </param>
+		/// <param name='guess'>
+		/// initial guess index.
+		/// </param>
+		/// <returns>
+		/// index of x with max(x) &lt;= target x or -1 if cannot find
+		/// </returns>
+		public static int FindBarIndex(IList<RectangleBarItem> items, double targetX, int guess)
```

Comment:
Yes, this is a problem that I contemplated upon. This method is performance-critical as it is called on every render event and it uses property getters excessively. I'm afraid that adding an additional function call to get a coordinate will impact performance. I haven't had the time to test it with performance profiler yet, I think I'll do it today. Maybe there is another way to make this method more generic so that I don't need to copy-paste it for every data item class.